src/server/pfs/server/driver.go

```diff
@@ -1915,26 +1911,30 @@ func isNotFoundErr(err error) bool {
 	return err != nil && strings.Contains(err.Error(), "not found")
 }
 
-func isNoHeadErr(err error) bool {
-	return errors.As(err, &pfsserver.ErrNoHead{})
-}
-
-func (d *driver) addBranchProvenance(branchInfo *pfs.BranchInfo, provBranch *pfs.Branch, sqlTx *sqlx.Tx) error {
+func (d *driver) addBranchProvenance(txnCtx *txncontext.TransactionContext, branchInfo *pfs.BranchInfo, provBranch *pfs.Branch) error {
 	if pfsdb.BranchKey(provBranch) == pfsdb.BranchKey(branchInfo.Branch) {
 		return errors.Errorf("provenance loop, branch %s cannot be provenant on itself", pfsdb.BranchKey(provBranch))
 	}
 	add(&branchInfo.Provenance, provBranch)
 	provBranchInfo := &pfs.BranchInfo{}
-	if err := d.branches.ReadWrite(sqlTx).Upsert(pfsdb.BranchKey(provBranch), provBranchInfo, func() error {
-		// Set provBranch, we may be creating this branch for the first time
-		provBranchInfo.Branch = provBranch
+	if err := d.branches.ReadWrite(txnCtx.SqlTx).Upsert(pfsdb.BranchKey(provBranch), provBranchInfo, func() error {
+		if provBranchInfo.Branch == nil {
```

Comment:
Does this mean that branches can be created through the provenance parameter in create branch?

Reply:
Yes - it looks like before it would lazily create provenance on a branch that didn't yet exist, but now we eagerly instantiate it.

Reply:
I guess this is fine, but it seems a bit sketchy to create the provenant branches if they don't already exist. If the prior code was not creating the branches and not erroring, then this is certainly preferable.